Chee Jer En iP

jerrrren opened the pull request:
# Jer En's PR review

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

- uses vscode
- uses prettier

1. I don't think there is much formatting issues since I used prettier😉
2. If you are reviewing my code try to focus more on naming

This are my **Todo**!

- [ ] Storing data in hard disk in date time format
- [ ] Simplify parse into more functions


***
This is the method to start the javafx application Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

jerrrren committed:
implement Greet Echo Exit

jerrrren committed:
implement Add and List

jerrrren committed:
implement Mark as Done

jerrrren committed:
Implemented ToDo,Event,Deadline

jerrrren committed:
Modify EXPECTED.txt

jerrrren committed:
Handle Errors

jerrrren committed:
implement delete

jerrrren committed:
add more exception

jerrrren committed:
fix bug in unmark case

jerrrren committed:
implement save

jerrrren committed:
implement save

jerrrren committed:
Delete duke.txt

jerrrren committed:
add data directory

jerrrren committed:
add create directory if does not exist

jerrrren committed:
implement string to date type

jerrrren committed:
Merge branch 'branch-Level-7'

jerrrren committed:
Merge branch 'branch-Level-8'

jerrrren committed:
add more oop

jerrrren committed:
add all to package duke

jerrrren committed:
Merge remote-tracking branch 'origin/add-gradle-support'

jerrrren committed:
add Gradle

jerrrren committed:
add Junit tests

jerrrren committed:
modify code to follow standard

jerrrren committed:
Merge branch 'CodingStandard'

jerrrren committed:
edit spacing in Parser.java

jerrrren committed:
add JavaDoc comments

jerrrren committed:
add find

jerrrren committed:
Merge branch 'branch-A-JavaDoc'

jerrrren committed:
Merge branch 'branch-Level-9'

jerrrren committed:
follow coding standards

alson001 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,218 @@
+    public boolean parse(String input, TaskList toDoList) throws DukeException {
+        String[] splitInput = input.split(" ");
+        String command = splitInput[0];
+        boolean isWritingToFile = false;
```
Doesn't isWrittentoFile sound better?

alson001 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,218 @@
+                String searchString = input.substring(("find").length() + 1);
+                ArrayList<Task> foundTasks = toDoList.search(searchString);
+                System.out.println("Here are the matching tasks in your list:");
+                for (int i = 0; i < foundTasks.size(); i++) {
```
```suggestion
                for (Task task : foundTasks) {
```

alson001 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,73 @@
+        return this.toString();
+    }
+
+    public Boolean contains(String searchString) {
```
Isn't shouldContains better here as 
Boolean variables/methods should be named to sound like booleans?

alson001 reviewed (commented):
Overall, I found your code easy to read. I noted a few minor coding standard violations
too. Consider moving System.out.printIn to UI as the code in Parser might be too long,

jerrrren committed:
add GUI functionality

jerrrren committed:
Merge branch 'branch-Level-10'

filbertphang commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,243 @@
+     * @return LocalDate object of parsed input string else null
+     */
+
+    private static LocalDate getDate(String[] splitInput) {
```
ok. personally would rename `splitInput` to `inputTokens` ("token" as in "tokenising" a string)

filbertphang commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,243 @@
+            }
+        }
+        return null;
+    }
```
should try to avoid [deep nesting](https://nus-cs2103-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting) or arrowhead-shaped code

would suggest you put the invalid cases at the top, then put the valid case at the bottom for better readability

```suggestion
        if (splitInput.length < 4) {
            // invalid case: <description>
            return null;
        } else if (!splitInput[splitInput.length - 2].equals("/by")) {
            // invalid case: <description>
            return null;
        } else {
            // do the thing
            ...
        }
```

filbertphang commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,243 @@
+        LocalDate taskDate;
+        StringBuilder response;
+
+        switch (command) {
```
coding standard violation: `case` statements should not be indented relative to `switch` statements, e.g.
```java
switch(myVar){
case 1:
    ...
    break
...
}
```

filbertphang replied:
i use vscode also but i don't use the prettier formatter. instead, i use the [java language support by redhat](https://marketplace.visualstudio.com/items?itemName=redhat.java) extension instead. 

you can clone [my formatter settings](https://github.com/filbertphang/ip/blob/master/.vscode/java-formatter.xml) if you want the target switch case behaviour. the relevant setting ID is `"org.eclipse.jdt.core.formatter.indent_switchstatements_compare_to_cases"`

filbertphang commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,243 @@
+
+                response.append("Here are the tasks in your list:\n");
+                for (int i = 0; i < toDoList.size(); i++) {
+
```
redundant blank line

filbertphang commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,98 @@
+package duke;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.FileWriter;
```
minor nitpick, maybe imports within the same package could be arranged alphabetically

filbertphang commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,98 @@
+
+
+public class Storage {
+    private String filePath;
```
naming is fine. to be completely unambiguous, you could make it `storageFilePath` but i think it's fine

filbertphang commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,98 @@
+    }
+
+    public Storage() {
+        this.filePath = "./data";
```
this is a magic string, you may want to extract it out to be some constant instead

filbertphang commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,98 @@
+    public void readStorage(ArrayList<Task> toDoList) {
+        try {
+            (new File(this.filePath)).mkdirs();
+            File savedFile = new File(this.filePath + "/duke.txt");
```
same thing here, the storage filename should probably be a constant too

filbertphang commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,73 @@
+        if (this.deadline == null) {
+            dateString = "";
+        } else {
+
```
another blank line

filbertphang commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,147 @@
+
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.control.Label;
```
another case where imports arent alphabetically ordered, but should be fine as long as you list your imports in a consistent manner

filbertphang commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,21 @@
+        String output;
+        try{
+            output = parser.parseInput("deadline hello /by 2019-10-15", new TaskList());
+        }catch(DukeException e){
```
add a space before `catch`

filbertphang reviewed (commented):
good work with the code. naming is generally fine, with minor nitpicks and no major suggestions. there are occasional violations of the coding standard but if you tweak your formatter enough those should go away (format on save is goated btw)

jerrrren committed:
add assertion to Parser.class parseInput method

jerrrren committed:
Apply SLAP to Parser and UI class and Storage class

The start method in UI class, the parser class parse method and
Storage class readStorage method all had multiple levels of abstraction

It needs to change as it violates SLAP principle and the code becomes excessively long

All the methods that violate SLAP principle has been abstracted out into individual methods for each
command for instance createEvent

Each command is given a method because it ensures the code length is sufficiently short and easy to understand

jerrrren committed:
Merge pull request #1 from jerrrren/branch-A-CodeQuality

Apply SLAP to Parser and UI class and Storage class

jerrrren committed:
add assertions

jerrrren committed:
Merge branch 'branch-A-assertions'

jerrrren committed:
add getupcoming deadlines

jerrrren committed:
modify bye command to exit application

jerrrren committed:
improve code quality

jerrrren committed:
improve comment quality

jerrrren committed:
abstract start method in Ui

jerrrren committed:
add photo

jerrrren committed:
rename photo

jerrrren committed:
Delete UI.png

jerrrren committed:
change name

jerrrren committed:
add UserGuide

jerrrren committed:
modify ip showcase

jerrrren committed:
add link to ug
